ocaml/xapi/test_vm_check_operation_error.ml

```diff
@@ -0,0 +1,43 @@
+
+let with_test_vm f =
+  let __context = Mock.make_context_with_new_db "Mock context" in
+  let vm_ref = Test_common.make_vm ~__context () in
+  f __context vm_ref
+
+(* CA-255128 Xapi_vm_lifecycle.check_operation_error should not throw an
+   exception when one of the nullable fields in the VM record are null. *)
```

Comment:
What I wanted to say is that the VM class has some fields that can validly be null in practice, i.e. it is not an error if they are null. And when one of these fields are null, the `check_operation_error` function should not fail.  When I create the test VM, the `Test_common.make_vm` function already sets many of the VM's fields to null, so I'm also testing other nullable fields here, not just the VDI. But I like your version, it's more concise, but I think we should still say that this test could fail due to other fields being null as well.